tests/testcases/mod.rs

```diff
@@ -68,6 +68,45 @@ pub mod all {
         tc.run(controller).await.unwrap()
     }
 
+    pub async fn key_value_validation_works(controller: &dyn Controller) {
+        async fn checks(
+            _: AppMetadata,
+            _: Option<BufReader<ChildStdout>>,
+            _: Option<BufReader<ChildStderr>>,
+        ) -> Result<()> {
+            Ok(())
+        }
+
+        let tc = TestCaseBuilder::default()
+            .name("key-value-undefined-store".to_string())
+            .appname(Some("key-value-undefined-store".to_string()))
+            .template(None)
+            .assertions(
+                |metadata: AppMetadata,
+                 stdout_stream: Option<BufReader<ChildStdout>>,
+                 stderr_stream: Option<BufReader<ChildStderr>>| {
+                    Box::pin(checks(metadata, stdout_stream, stderr_stream))
+                },
+            )
+            .build()
+            .unwrap();
+
+        let e = tc.run(controller).await.unwrap_err();
+        let err_text = format!("{e:#}");
+
+        let expected1 = "Component hello is granted access to key-value store 'anaspeptic', which is not defined";
+        let expected2 = "Component hello is granted access to key-value store 'pericombobulations', which is not defined";
```

Comment:
If we want to test for specific UI text we should consider "golden file" testing which semi-automates updating this kind of assertion: https://crates.io/crates/goldenfile

Reply:
Agree but I propose to declare this out of scope for this PR.